pkg/client/unversioned/discovery_client.go

```diff
@@ -116,21 +101,17 @@ func (d *DiscoveryClient) ServerGroups() (apiGroupList *unversioned.APIGroupList
 
 // ServerResourcesForGroupVersion returns the supported resources for a group and version.
 func (d *DiscoveryClient) ServerResourcesForGroupVersion(groupVersion string) (resources *unversioned.APIResourceList, err error) {
-	url := d.baseURL
+	url := url.URL{}
 	if groupVersion == "v1" {
 		url.Path = "/api/" + groupVersion
 	} else {
 		url.Path = "/apis/" + groupVersion
 	}
-	resp, err := d.get(url.String())
+	resources = &unversioned.APIResourceList{}
+	err = d.Get().AbsPath(url.String()).Do().Into(resources)
```

Comment:
>If an error happens, does this get our nice status error type?

Yes.

v1.1 server does not encode Status, so the unmarshal the json will decode it.

in v1.0, Status is in v1 types.go, server will encode Status to v1. jsonCodec will decode it to unversioned.Status and ignore the TypeMeta. So it still works.